MSALiOSB2C/ViewController.swift

```diff
@@ -228,123 +196,94 @@ class ViewController: UIViewController, UITextFieldDelegate, URLSessionDelegate
              flow completes, or encounters an error.
              */
             
-            let thisUser = try self.getUserByPolicy(withUsers: application.users(), forPolicy: kSignupOrSigninPolicy)
-    
-            application.acquireTokenSilent(forScopes: kScopes, user: thisUser ) { (result, error) in
-                    if error == nil {
-                        self.accessToken = (result?.accessToken)!
-                        self.loggingText.text = "Refreshing token silently"
-                        self.loggingText.text = "Refreshed Access token is \(self.accessToken)"
-                        
-                    }  else if (error! as NSError).code == MSALErrorCode.interactionRequired.rawValue {
-                        
-                        // Notice we supply the user here. This ensures we acquire token for the same user
-                        // as we originally authenticated.
-                        
-                        application.acquireToken(forScopes: self.kScopes, user: thisUser) { (result, error) in
-                                if error == nil {
-                                    self.accessToken = (result?.accessToken)!
-                                    self.loggingText.text = "Access token is \(self.accessToken)"
-                                    
-                                } else  {
-                                    self.loggingText.text = "Could not acquire new token: \(error ?? "No error informarion" as! Error)"
-                                }
-                        }
-                    } else {
-                        self.loggingText.text = "Could not acquire token: \(error ?? "No error informarion" as! Error)"
-                    }
+            guard let thisAccount = try self.getAccountByPolicy(withAccounts: application.allAccounts(), policy: kSignupOrSigninPolicy) else {
+                self.loggingText.text = "There is no account available!"
+                return
             }
-            } catch {
-            self.loggingText.text = "Unable to create application \(error)"
             
+            let parameters = MSALSilentTokenParameters(scopes: kScopes, account:thisAccount)
+            parameters.authority = authority
+            self.application.acquireTokenSilent(with: parameters) { (result, error) in
+                if error == nil {
+                    self.accessToken = (result?.accessToken)!
+                    self.loggingText.text = "Refreshing token silently"
+                    self.loggingText.text = "Refreshed Access token is \(self.accessToken)"
+                    
+                }  else if ((error! as NSError).code == MSALError.interactionRequired.rawValue) {
```

Comment:
nit: can you please try to avoid using "!"? It's generally seen as a bit risky practice. 

Here's a sample from AAD MSAL sample for the same handling avoiding !

```
if let error = error {
                
                let nsError = error as NSError
                
                // interactionRequired means we need to ask the user to sign-in. This usually happens
                // when the user's Refresh Token is expired or if the user has changed their password
                // among other possible reasons.
                
                if (nsError.domain == MSALErrorDomain) {
                    
                    if (nsError.code == MSALError.interactionRequired.rawValue) {
                        
                        DispatchQueue.main.async {
                            self.acquireTokenInteractively()
                        }
                        return
                    }
                }
                
                self.updateLogging(text: "Could not acquire token silently: \(error)")
                return
            }
            
            guard let result = result else {
                
                self.updateLogging(text: "Could not acquire token: No result returned")
                return
            }
            
            self.accessToken = result.accessToken
            self.updateLogging(text: "Refreshed Access token is \(self.accessToken)")
            self.updateSignOutButton(enabled: true)
            self.getContentWithToken()
```

Reply:
Nice, I learned :)